cli/src/pcluster/cluster_model.py

```diff
@@ -115,17 +117,82 @@ def _ec2_run_instance(self, pcluster_config, **kwargs):  # noqa: C901 FIXME!!!
+        """Return the id of the current official image, for the provided os-architecture combination."""
+        ami_id = None
+        try:
+            image_prefix = self._get_official_image_name_prefix(os, architecture)
+            # Look for public ParallelCluster AMI in released version
+            ami_id = self._get_image_id(image_prefix, "amazon", True)
+            if not ami_id:
+                # Look for dev account owner during development
+                ami_id = self._get_image_id(image_prefix, "self", False)
+        except ClientError as e:
+            raise Exception(
+                "Unable to retrieve official image id for base_os='{0}' and architecture='{1}': {2}".format(
+                    os, architecture, e.response.get("Error").get("Message")
+                )
+            )
+        if not ami_id:
+            raise Exception(
+                "No official image id found for base_os='{0}' and architecture='{1}'".format(os, architecture)
+            )
+
+        return ami_id
+
+    def _get_image_id(self, image_prefix, owner, public=False):
+        """Search for the ami with the specified prefix from the given account and returns the id if found."""
+        filters = [
+            {
+                "Name": "name",
+                "Values": ["{0}*".format(image_prefix)],
+            }
+        ]
+        if public:
+            filters.append({"Name": "is-public", "Values": ["true"]})
+
+        images = boto3.client("ec2").describe_images(Filters=filters, Owners=[owner]).get("Images")
+        return images[0].get("ImageId") if images else None
+
+    def _get_official_image_name_prefix(self, os, architecture):
+        """Return the prefix of the current official image, for the provided os-architecture combination."""
+        suffixes = {
+            "alinux": "amzn-hvm",
+            "alinux2": "amzn2-hvm",
+            "centos7": "centos7-hvm",
+            "centos8": "centos8-hvm",
+            "ubuntu1604": "ubuntu-1604-lts-hvm",
+            "ubuntu1804": "ubuntu-1804-lts-hvm",
+        }
+        return "aws-parallelcluster-{version}-{suffix}-{arch}".format(
+            version=get_installed_version(), suffix=suffixes[os], arch=architecture
+        )
+
+    def _get_cluster_ami_id(self, pcluster_config):
+        """Get the image id of the cluster."""
+        cluster_ami = None
+        os = pcluster_config.get_section("cluster").get_param_value("base_os")
+        architecture = pcluster_config.get_section("cluster").get_param_value("architecture")
+        custom_ami = pcluster_config.get_section("cluster").get_param_value("custom_ami")
+        try:
+            cluster_ami = custom_ami if custom_ami else self._get_official_image_id(os, architecture)
+        except Exception as e:
+            pcluster_config.error("Error when resolving cluster ami id: {0}".format(e))
+
+        return cluster_ami
+
+    def _get_latest_alinux_ami_id(self):
```

Comment:
shall we get rid of this unused function

Reply:
:+1: